cli/tools/repl/session.rs

```diff
@@ -408,3 +456,44 @@ impl ReplSession {
       .and_then(|res| serde_json::from_value(res).map_err(|e| e.into()))
   }
 }
+
+/// Walk an AST and get all import specifiers for analysis if any of them is
+/// an NPM specifier.
+pub struct ImportCollector {
+  pub imports: Vec<String>,
+}
+
+impl ImportCollector {
+  pub fn new() -> Self {
+    Self { imports: vec![] }
+  }
+}
+
+impl Visit for ImportCollector {
+  noop_visit_type!();
+
+  fn visit_call_expr(&mut self, call_expr: &swc_ast::CallExpr) {
+    if !matches!(call_expr.callee, swc_ast::Callee::Import(_)) {
+      return;
+    }
+
+    if !call_expr.args.is_empty() {
+      let arg = &call_expr.args[0];
+      if let swc_ast::Expr::Lit(swc_ast::Lit::Str(str_lit)) = &*arg.expr {
+        self.imports.push(str_lit.value.to_string());
+      }
+    }
+  }
+
+  fn visit_module_decl(&mut self, module_decl: &swc_ast::ModuleDecl) {
+    use deno_ast::swc::ast::*;
+
+    if let ModuleDecl::Import(import_decl) = module_decl {
```

Comment:
What happens when someone does `export {} from "npm:chalk";`? I know it's strange, but we should still collect them. I believe it gets transpiled to a dynamic import that imports for the side effects.

Reply:
Good catch:
```
> export {} from "npm:chalk";
Uncaught TypeError: Could not resolve 'npm:chalk'.
    at async <anonymous>:2:1
```